tests/pyfunc/test_model_export_with_class_and_artifacts.py

```diff
@@ -1308,48 +1252,41 @@ def test_functional_python_model_list_dict_to_list_without_example(tmp_path):
         path=tmp_path, python_model=list_dict_to_list, pip_requirements=["pandas"]
     )
     model = Model.load(tmp_path)
-    assert model.signature.inputs.to_dict() == [{"type": "string", "required": True}]
-    assert model.signature.outputs.to_dict() == [{"type": "string", "required": True}]
+    assert model.signature.inputs == Schema([ColSpec(Array(Map("string")))])
+    assert model.signature.outputs == Schema([ColSpec(Array("string"))])
     loaded_model = mlflow.pyfunc.load_model(tmp_path)
     assert loaded_model.predict([{"a": "x"}, {"a": "y"}]) == ["ax", "ay"]
 
 
 @pytest.mark.parametrize(
-    ("input_example", "expected_error_message"),
+    ("input_example"),
     [
-        ([], "non-empty"),
-        ([0], "to be string"),
-        ([{"a": "b"}], "to be string"),
+        [0],
+        [{"a": "b"}],
     ],
 )
-def test_functional_python_model_list_invalid_example(
-    tmp_path, input_example, expected_error_message
-):
-    with pytest.raises(MlflowException, match=expected_error_message):
+def test_functional_python_model_list_invalid_example(tmp_path, input_example):
+    with mock.patch("mlflow.models.signature._logger.warning") as mock_warning:
         mlflow.pyfunc.save_model(
             path=tmp_path, python_model=list_to_list, input_example=input_example
         )
+        assert "Input example is not compatible with the type hint" in mock_warning.call_args[0][0]
 
 
 @pytest.mark.parametrize(
-    ("input_example", "expected_error_message"),
+    "input_example",
     [
-        ([], "non-empty"),
-        (["a"], "to be dict"),
-        ([{}], "at least one item"),
-        ([{0: "a"}], "string keys"),
-        ([{"a": 0}], "string values"),
-        ([{"a": "x"}, {"b": "y"}], r"dict with keys \['a'\]"),
-        ([{"a": "x"}, {"a": "y", "b": "z"}], r"dict with keys \['a'\]"),
+        ["a"],
+        [{0: "a"}],
+        [{"a": 0}],
     ],
 )
-def test_functional_python_model_list_dict_invalid_example(
-    tmp_path, input_example, expected_error_message
-):
-    with pytest.raises(MlflowException, match=expected_error_message):
+def test_functional_python_model_list_dict_invalid_example(tmp_path, input_example):
+    with mock.patch("mlflow.models.signature._logger.warning") as mock_warning:
         mlflow.pyfunc.save_model(
             path=tmp_path, python_model=list_dict_to_list, input_example=input_example
         )
+        assert "Input example is not compatible with the type hint" in mock_warning.call_args[0][0]
```

Comment:
This line would break if `_logger.warning` is called with another warning. Can we use `any(... for call_args in mock_warning.call_args_list)`?